flink-runtime/src/test/java/org/apache/flink/runtime/dispatcher/TestingJobManagerRunnerRegistry.java

```diff
@@ -118,6 +122,93 @@ public static Builder builder() {
+                                        () -> false))
+                .withGetFunction(
+                        jobId ->
+                                accessReference(
+                                        singleRunnerReference,
+                                        Function.identity(),
+                                        () -> throwNoSuchElementException(jobId)))
+                .withGetJobManagerRunnersSupplier(
+                        () -> CollectionUtil.ofNullable(singleRunnerReference.get()))
+                .withSizeSupplier(() -> singleRunnerReference.get() == null ? 0 : 1)
+                .withGetRunningJobIdsSupplier(
+                        () ->
+                                accessReference(
+                                        singleRunnerReference,
+                                        jobManagerRunner ->
+                                                Collections.singleton(jobManagerRunner.getJobID()),
+                                        Collections::emptySet))
+                .withUnregisterFunction(
+                        jobId ->
+                                accessReference(
+                                        singleRunnerReference,
+                                        jobManagerRunner -> {
+                                            if (jobManagerRunner.getJobID().equals(jobId)) {
+                                                return singleRunnerReference.getAndSet(null);
+                                            }
+
+                                            return throwNoSuchElementException(jobId);
+                                        },
+                                        () -> throwNoSuchElementException(jobId)))
+                .withGlobalCleanupAsyncFunction(
+                        (actualJobId, ignoredExecutor) ->
+                                cleanup(singleRunnerReference, actualJobId))
+                .withLocalCleanupAsyncFunction(
+                        (actualJobId, executor) -> cleanup(singleRunnerReference, actualJobId));
+    }
+
+    private static CompletableFuture<Void> cleanup(
+            AtomicReference<JobManagerRunner> singleRunnerReference, JobID actualJobId) {
+        return accessReference(
+                singleRunnerReference,
+                jobManagerRunner -> {
+                    if (jobManagerRunner.getJobID().equals(actualJobId)) {
+                        singleRunnerReference.getAndSet(null).closeAsync();
+                    }
+
+                    return FutureUtils.completedVoidFuture();
+                },
+                FutureUtils::completedVoidFuture);
+    }
+
+    private static JobManagerRunner throwNoSuchElementException(JobID jobId) {
+        throw new NoSuchElementException(
+                "JobManagerRunner with job ID " + jobId + " is not registered.");
+    }
+
+    private static <T> T accessReference(
```

Comment:
```suggestion
    private static <T> T applyToReferenceOrFallback(
```

Reply:
This method got removed using the `Optional.ofNullable` approach.